### PR TITLE
Added Weighted Unrealised Reward fn

### PR DESCRIPTION
**Pull request by arunavo4**
@notadamking Added Weighted Unrealised Reward Function. This is the best way to reward a agent in our case, that I have found in my experience.

### CODE REVIEW

File: lib/env/TradingEnv.py [hunk 1]
```diff
@@ -101,25 +120,11 @@ def _take_action(self, action):
...
-        reward = self.reward_fn(observations=self.observations,
-                                net_worths=self.net_worths,
-                                account_history=self.account_history,
-                                last_bought=self.last_bought,
-                                last_sold=self.last_sold,
-                                current_price=self._current_price())
+        if self.account_history['btc_sold'].values[-1] > 0:
+            reward = self.get_reward(self.account_history['revenue_from_sold'].values[-1])
+        else:
+            reward = self.get_reward(self.btc_held * self._current_price())
```
Comment (raybotha) [inline lib/env/TradingEnv.py]: You probably want to call the `self.reward_fn` here

Comment (arunavo4) [reply lib/env/TradingEnv.py]: Ohh ok I will make the changes

Comment (arunavo4) [reply lib/env/TradingEnv.py]: I have updated the code.

**Comment by notadamking** [comment]
The current `TradingEnv` supports any `reward_fn` to be passed in as `kwargs`. All of the necessary data you need to calculate the weighted unrealized reward should already be available to you in the `net_worths` and `account_history` arguments of `reward_fn`.

A better solution (needed for me to review the PR), would be to create a new directory `lib/rewards` and a new file `lib/rewards/weighted_unrealized.py`, which implements the `reward_fn` interface outside of the scope of `TradingEnv`.

**Comment by arunavo4** [comment]
Great I will try to work it out .

**Pull request closed without merging**